app/scripts/services/dimBungieService.factory.js

```diff
@@ -493,27 +493,29 @@
     }
 
     function getDestinyVendors(token, platform, membershipId, characters) {
+      // sparrowHash = 1501155019,
       // Titan van, Hunter van, Warlock van, Van quart, Dead orb, Future war, New mon, Eris Morn, Cruc hand, Cruc quart, Speaker, Variks, Exotic Blue, Banner
-      var vendorHashes = ['1990950', '3003633346', '1575820975', '2668878854', '3611686524', '1821699360', '1808244981', '174528503', '3746647075', '3658200622', '2680694281', '1998812735', '3902439767', '242140165'];
-      var promises = [];
-      _.each(vendorHashes, function(vendorHash) {
-        _.each(characters, function(character) {
-          var processPB = processVendorsResponse.bind(null, character);
-          promises.push(
-            $q.when(getCharacterVendorsRequest(token, platform, membershipId, character, vendorHash))
-              .then($http)
-              .then(handleErrors)
-              .then(processPB)
-              .catch(function(e) {
-                if (e.message !== 'The Vendor you requested was not found.') {
-                  throw e;
-                }
-              })
-          );
+      // var vendorHashes = ['1990950', '3003633346', '1575820975', '2668878854', '3611686524', '1821699360', '1808244981', '174528503', '3746647075', '3658200622', '2680694281', '1998812735', '3902439767', '242140165'];
+      return $q.when(dimVendorDefinitions).then(function(vendors) {
+        var promises = [];
+        _.each(_.keys(vendors), function(vendorHash) {
```

Comment:
You don't need to run `_.keys` - `_.each`'s callback is `function(value, key)`: http://underscorejs.org/#each